packages/wallets/src/components/WalletTourGuide/WalletTourGuide.tsx

```diff
@@ -51,24 +67,42 @@ const WalletTourGuide = () => {
 
         if (action === ACTIONS.RESET) {
             setWalletsOnboarding('');
+            setRun(false);
             if (!isAllWalletsAlreadyAdded && addMoreWalletRef.current) {
                 addMoreWalletRef.current.style.transform = addMoreWalletsTransformValue;
             }
         }
     };
 
     useEffect(() => {
-        const switchToFiatWallet = () => {
+        const needToStart = walletsOnboarding === START_VALUE;
+        if (needToStart) {
+            const isAnythingLoading =
+                dxtradeIsLoading ||
+                ctraderIsLoading ||
+                isFetching ||
+                !isSuccess ||
+                isLoading ||
+                sorteAccountsListLoading;
             if (fiatWalletLoginId && fiatWalletLoginId !== activeWalletLoginId) {
                 switchWalletAccount(fiatWalletLoginId);
+            } else if (needToStart && !isAnythingLoading) {
+                setRun(true);
+                setWalletsOnboarding('');
             }
-        };
-
-        const needToStart = walletsOnboarding === startValue;
-        if (needToStart) {
-            switchToFiatWallet();
         }
-    }, [activeWalletLoginId, fiatWalletLoginId, switchWalletAccount, walletsOnboarding]);
+    }, [
+        activeWalletLoginId,
+        fiatWalletLoginId,
+        switchWalletAccount,
+        walletsOnboarding,
+        isLoading,
+        isFetching,
+        isSuccess,
+        dxtradeIsLoading,
+        ctraderIsLoading,
+        sorteAccountsListLoading,
+    ]);
```

Comment:
Seems like there is a missing dependency `setWalletsOnboarding` for this useEffect. Is that not needed?

Reply:
updated PR and removed this fragment actually,